pkg/controller/certificates/trigger/trigger_controller_test.go

```diff
@@ -312,6 +315,65 @@ func TestProcessItem(t *testing.T) {
 				},
 			},
 		},
+		"should set the Issuing condition if the last failure is set but the is a mismatch between the certificate and request": {
+			certificate: gen.Certificate("test",
+				gen.SetCertificateNamespace("testns"),
+				gen.SetCertificateUID(types.UID("test-uid")),
+				gen.SetCertificateDNSNames("example.com"),
+				gen.SetCertificateRevision(1),
+				gen.SetCertificateLastFailureTime(metav1.NewTime(now)),
+			),
+			requests: []*cmapi.CertificateRequest{createCertificateRequestOrPanic(
+				gen.Certificate("test",
+					gen.SetCertificateNamespace("testns"),
+					gen.SetCertificateUID(types.UID("test-uid")),
+					gen.SetCertificateDNSNames("example2.com"), // Mismatch here.
+					gen.SetCertificateRevision(1),
+					gen.SetCertificateLastFailureTime(metav1.NewTime(now)),
+				)),
+			},
+			chainShouldEvaluate:        true,
+			chainShouldTriggerIssuance: true,
+			expectedEvent:              "Normal Issuing Re-issuance forced by unit test case",
+			expectedConditions: []cmapi.CertificateCondition{
+				{
+					Type:               cmapi.CertificateConditionIssuing,
+					Status:             cmmeta.ConditionTrue,
+					Reason:             forceTriggeredReason,
+					Message:            forceTriggeredMessage,
+					LastTransitionTime: &metaNow,
+				},
+			},
+		},
+		"should set the Issuing condition if the last failure is set and that the revision has not be set yet": {
```

Comment:
maybe make sure unit test names reflect that we are testing
- failing = true && mismatch == true
- failing = true && mismatch == false
- failing = false && mismatch == false
- failing = true && mismatch == true

Reply:
Closing for now, I'll try to find a way to make the test cases "more explicit"